### DISCUSSION TITLE
How to use Xml file in ResourceInclude?

### DISCUSSION THREAD

**Discussion by Dennis-Christian** [AvaloniaUI/Avalonia#7390] [Q&A]
Hello,
i built an app which supports localization via xml files with key/value pairs that are loaded into the MergedDictionaries of the Current Application and used via DynamicResource Bindings in c# + WPF.

Now i want to port this app to Avalonia but encountered a few hindrances, maybe you could help me:

The xml file looks like this:

```
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                                 xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" 
                                 xmlns:system="clr-namespace:System;assembly=System.Runtime">
	<system:String x:Key="lbl_dashboarde">Dashboard</system:String>
	<system:String x:Key="lbl_manual">Manual</system:String>
	<system:String x:Key="lbl_copyrights">Copyrights</system:String>
	<system:String x:Key="lbl_write">Write</system:String>
</ResourceDictionary>
```

And the code like this:

```
void SetLanguageDictionary()
{
    if (selectedLanguage == null) return;

    Uri uri;
    if (selectedLanguage.File.StartsWith("avares://"))
    {
        uri = new Uri(selectedLanguage.File);
    }
    else
    {
        string assemblyName = Assembly.GetEntryAssembly().GetName().Name;
        uri = new Uri($"avares://{assemblyName}{selectedLanguage.File}");
    }

    var resInclude = new ResourceInclude();
    resInclude.Source = uri;

    Application.Current.Resources.MergedDictionaries.Add(resInclude); //<- Here happens the XamlLoadException
}
```

This is the Uri that is generated `avares://toolstar_ImageWriter/Assets/localization.xml`
File exists and is added to assets as an AvaloniaResource

The Exception says `XamlLoadException: No precompiled XAML found for avares://toolstar_ImageWriter/Assets/localization.xml (baseUri: ), make sure to specify x:Class and include your XAML file as AvaloniaResource`

What am i missing?

Thanks for your help!

**Comment by ahopper** [comment]
what happens if you rename it to .axaml ?

**Reply by Dennis-Christian** [reply]
I renamed the file as you suggested and changed first xml namespace to `xmlns="https://github.com/avaloniaui"`

No change, same error

**Comment by ahopper** [comment]
This loaded for me with no error
```
<ResourceDictionary xmlns="https://github.com/avaloniaui"
                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
					xmlns:system="clr-namespace:System;assembly=System.Runtime">
    <system:String x:Key="lbl_dashboarde">Dashboard</system:String>
	<system:String x:Key="lbl_manual">Manual</system:String>
	<system:String x:Key="lbl_copyrights">Copyrights</system:String>
	<system:String x:Key="lbl_write">Write</system:String>
</ResourceDictionary>
```
saved as .axaml
